include/dfu/dfu_target_uart.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ * Copyright (c) 2022 INTELLINIUM <giuliano.franchetto@intellinium.com>
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef DFU_TARGET_UART_H__
+#define DFU_TARGET_UART_H__
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <dfu/dfu_target.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ *  @def DFU_UART_HEADER_MAGIC
+ *
+ *  @brief UART target magic value. This value needs to be set in the envelop to
+ *  select UART as active DFU type
+ */
+#define DFU_UART_HEADER_MAGIC               0x85f3d83a
+
+/**
+ * @struct dfu_target_uart_remote
+ *
+ * @brief Structure representing a single remote endpoint which can receive
+ * application update
+ */
+struct dfu_target_uart_remote {
+	/**
+	 * Target identifier.
+	 * This value shall be used in the cbor envelop
+	 */
+	uint8_t identifier;
+
+	/**
+	 * The uart device to use.
+	 * Must not be NULL
+	 */
+	const struct device *device;
+};
+
+/**
+ * @struct dfu_target_uart_params
+ *
+ * @brief Structure representing the configuration of the uart target type
+ */
+struct dfu_target_uart_params {
+	/**
+	 * the number of remotes in the configuration
+	 */
+	size_t remote_len;
```

Comment:
`remotes_len` or `remote_count`; this sounds like length of single remote, whatever remote would be.

Reply:
done